tests/test_macid.py

```diff
@@ -92,26 +103,44 @@ def test_get_all_pure_spe(self) -> None:
         self.assertTrue(np.array_equal(cpd_d2.values, np.array([[0, 1], [1, 0]])))
 
         macid = modified_taxi_competition()
-        all_spe = macid.get_all_pure_spe()
+        all_spe = macid.get_all_spe()
         self.assertTrue(len(all_spe) == 2)
 
         macid = prisoners_dilemma()
-        all_spe = macid.get_all_pure_spe()
+        all_spe = macid.get_all_spe()
         self.assertTrue(len(all_spe) == 1)
 
         macid = battle_of_the_sexes()
-        all_spe = macid.get_all_pure_spe()
+        all_spe = macid.get_all_spe()
         self.assertTrue(len(all_spe) == 2)
 
         macid3 = basic_different_dec_cardinality()
-        all_spe = macid3.get_all_pure_spe()
+        all_spe = macid3.get_all_spe()
         spe = all_spe[0]
         joint_policy = macid3.policy_profile_assignment(spe)
         cpd_d1 = joint_policy["D1"]
         cpd_d2 = joint_policy["D2"]
         self.assertTrue(np.array_equal(cpd_d1.values, np.array([0, 1])))
         self.assertTrue(np.array_equal(cpd_d2.values, np.array([[0, 0], [1, 0], [0, 1]])))
 
+        macid = battle_of_the_sexes()
+        all_spe = macid.get_all_spe(solver="enummixed")
+        self.assertTrue(len(all_spe) == 3)
+        macid = prisoners_dilemma()
+        all_spe = macid.get_all_spe(solver="enummixed")
+        self.assertTrue(len(all_spe) == 1)
+        macid = taxi_competition()
+        all_spe = macid.get_all_spe(solver="enummixed")
+        self.assertTrue(len(all_spe) == 1)
+        macid = modified_taxi_competition()
+        all_spe = macid.get_all_spe(solver="enummixed")
+        self.assertTrue(len(all_spe) == 2)
```

Comment:
for spe with solver other than enumpure, this will only return 1 (potentially mixed) spe because the solver will default to a mixed solver for non-2-player games. So here there are 2 pure spes but it will only return one. It will return both if solver="enumpure". Is this desired/expected behaviour?

Reply:
hmm okay I think we want it to default to find as many NE as possible. Therefore, maybe it should check how many players there are first
- if it's a two player game, then it will default to a solver that finds all NE (pure and mixed)
- if it's got > 2 players, then it will find all pure  and then append on any mixed NE it finds if this differs from the pure NE already found